diff -Naur ./gcc/c89 ../gcc/c89
--- ./gcc/c89	2023-07-29 09:24:01.205619856 +0000
+++ ../gcc/c89	2023-07-29 09:22:31.813129172 +0000
@@ -8,3 +8,4 @@
   esac
 done
 exec gcc $fl ${1+"$@"}
+#change here
diff -Naur ./gcc/PKGBUILD ../gcc/PKGBUILD
--- ./gcc/PKGBUILD	2023-07-29 09:24:01.205619856 +0000
+++ ../gcc/PKGBUILD	2023-07-29 09:22:37.325159430 +0000
@@ -3,7 +3,7 @@
 # Contributor:  Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
 # Contributor: Allan McRae <allan@archlinux.org>
 # Contributor: Daniel Kozak <kozzi11@gmail.com>
-
+#helo there
 # toolchain build order: linux-api-headers->glibc->binutils->gcc->glibc->binutils->gcc
 # NOTE: libtool requires rebuilt with each new gcc version
 
