diff --git a/PKGBUILD b/PKGBUILD
index a362b64..a3dc8e9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,16 +32,16 @@ validpgpkeys=(38A644698C69787344E954CE29EE848AE2CCF3F4) # Andrew G. Morgan <morg
 
 build() {
   local make_options=(
-    CC="gcc -m32"
+    CC="armv7h-linux-gnueabihf-gcc"
     DYNAMIC=yes
-    KERNEL_HEADERS=/usr/include
+    KERNEL_HEADERS=/usr/include32
     lib=lib32
     prefix=/usr
     sbindir=bin
     -C $_name-$pkgver
   )
 
-  make "${make_options[@]}"
+  make "${make_options[@]}" -j$(nproc)
 }
 
 check() {
@@ -53,6 +53,7 @@ package() {
     DESTDIR="$pkgdir"
     RAISE_SETFCAP=no
     lib=lib32
+    headers=include32
     prefix=/usr
     sbindir=bin
     install
@@ -66,6 +67,8 @@ package() {
   make "${make_options[@]}"
   install -vDm 644 $_name-$pkgver/{CHANGELOG,README} -t "$pkgdir/usr/share/doc/$pkgname/"
   install -vDm 644 $_name-$pkgver/License -t "$pkgdir/usr/share/licenses/$pkgname/"
+  mv -v "$pkgdir/usr/include" "$pkgdir/usr/include32"
+  sed -i 's|/usr/include|/usr/include32|g' "$pkgdir/usr/lib32/pkgconfig/"*.pc
   # remove files provided by libcap
-  rm -rv "$pkgdir/usr/"{include,share/man,bin}
+  rm -rv "$pkgdir/usr/"{share/man,bin}
 }
