diff --git a/PKGBUILD b/PKGBUILD
index 6875f69..59c30a6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -35,9 +35,9 @@ validpgpkeys=(
 
 build() {
   local make_options=(
-    CC="gcc -m32"
+    CC="armv7h-linux-gnueabihf-gcc"
     DYNAMIC=yes
-    KERNEL_HEADERS=/usr/include
+    KERNEL_HEADERS=/usr/include32
     lib=lib32
     prefix=/usr
     sbindir=bin
@@ -48,7 +48,7 @@ build() {
 }
 
 check() {
-  make test CC="gcc -m32" -k -C $_name
+  make test CC="armv7h-linux-gnueabihf-gcc" -k -C $_name
 }
 
 package() {
@@ -56,6 +56,7 @@ package() {
     DESTDIR="$pkgdir"
     RAISE_SETFCAP=no
     lib=lib32
+    headers=include32
     prefix=/usr
     sbindir=bin
     install
@@ -70,6 +71,8 @@ package() {
   make "${make_options[@]}"
   install -vDm 644 $_name/{CHANGELOG,README} -t "$pkgdir/usr/share/doc/$pkgname/"
   install -vDm 644 $_name/License -t "$pkgdir/usr/share/licenses/$pkgname/"
+  mv -v "$pkgdir/usr/include" "$pkgdir/usr/include32"
+  sed -i 's|/usr/include|/usr/include32|g' "$pkgdir/usr/lib32/pkgconfig/"*.pc
   # remove files provided by libcap
-  rm -rv "$pkgdir/usr/"{include,share/man,bin}
+  rm -rv "$pkgdir/usr/"{share/man,bin}
 }
