diff --git i/PKGBUILD w/PKGBUILD
index 638c6e0..297ec16 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -40,8 +40,13 @@ prepare() {
 }
 
 build() {
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
+
   local meson_options=(
-    --cross-file lib32
+    --libdir=/usr/lib32
+    --includedir=/usr/include32
     -D docs=disabled
     -D icu=enabled
     -D legacy=enabled
@@ -59,9 +64,14 @@ check() {
 package_lib32-libxml2() {
   provides=(libxml2.so)
 
+  local meson_options=(
+      --libdir=/usr/lib32
+      --includedir=/usr/include32
+  )
+
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir"/usr/{bin,include}
+  rm -r "$pkgdir"/usr/bin
 
   install -Dm644 libxml2/Copyright -t "$pkgdir/usr/share/licenses/$pkgname"
 }
