diff --git a/PKGBUILD b/PKGBUILD
index 916758e..7670bce 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,7 +39,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D doc=disabled
   )
 
@@ -60,7 +60,7 @@ package() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir"/{etc,usr/{include,share},var}
+  rm -r "$pkgdir"/{etc,usr/share,var}
 
   mv "$pkgdir"/usr/bin/fc-cache{,-32}
   find "$pkgdir"/usr/bin -type f -not -name '*-32' -delete
