diff --git a/PKGBUILD b/PKGBUILD
index 15e7289..65d27bb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,7 +50,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D default_library=shared
   )
 
@@ -67,7 +67,7 @@ package_lib32-freetype2() {
 
   meson install -C build --destdir="$pkgdir"
 
-  rm -r "$pkgdir"/usr/{include,share}
+  rm -r "$pkgdir"/usr/share
 
   install -Dm644 freetype-$pkgver/docs/FTL.TXT \
     -t "$pkgdir/usr/share/licenses/$pkgname"
