diff --git a/PKGBUILD b/PKGBUILD
index b599b4a..b83fde9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,6 +28,7 @@ makedepends=(
   git
   meson
   valgrind
+  lib32-xorgproto
 )
 _commit=3909090108bb2db55330e3eb148aebe664735363  # tags/1.18.0^0
 source=("git+https://gitlab.freedesktop.org/cairo/cairo.git#commit=$_commit")
@@ -44,7 +45,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D dwrite=disabled
     -D gtk_doc=false
     -D spectre=disabled
@@ -53,6 +54,7 @@ build() {
   )
 
   arch-meson cairo build "${meson_options[@]}"
+  sed -i 's|-I/usr/include |-I/usr/include32 |' build/build.ninja
   meson compile -C build
 }
 
@@ -65,7 +67,7 @@ package_lib32-cairo() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir"/usr/{include,bin}
+  rm -r "$pkgdir"/usr/bin
 }
 
 # vim:set sw=2 sts=-1 et:
