diff --git a/PKGBUILD b/PKGBUILD
index e8264b3..93c08cf 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,6 +30,7 @@ build() {
     -D BUILD_SHARED_LIBS=ON
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_LIBDIR=lib32
+    -D CMAKE_INSTALL_INCLUDEDIR=include32
     -D CMAKE_INSTALL_PREFIX=/usr
     -D CMAKE_INSTALL_SYSCONFDIR=/etc
     -D CMAKE_SKIP_INSTALL_RPATH=ON
@@ -38,12 +39,12 @@ build() {
     -D SPIRV_WERROR=OFF
   )
 
-  export ASFLAGS+=" --32"
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cmake -S SPIRV-Tools -B build -G Ninja "${cmake_options[@]}"
+  sed -i 's|-I/usr/include|-I/usr/include32|' build/build.ninja
   cmake --build build
 }
 
@@ -53,7 +54,7 @@ check() {
 
 package() {
   DESTDIR="$pkgdir" cmake --install build
-  rm -r "$pkgdir"/usr/{bin,include}
+  rm -r "$pkgdir"/usr/bin
 }
 
 # vim:set sw=2 sts=-1 et:
