diff --git a/PKGBUILD b/PKGBUILD
index 2e71d00..c366357 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,7 +46,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D default_bus=dbus-broker
     -D docs=false
     -D gtk2_atk_adaptor=false
@@ -80,7 +80,8 @@ package_lib32-at-spi2-core() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "${pkgdir}"/{etc,usr/{include,lib,share}}
+  mv "${pkgdir}"/usr/lib "${pkgdir}"/usr/lib32
+  rm -r "${pkgdir}"/{etc,usr/share}
 }
 
 # vim:set sw=2 sts=-1 et:
