diff --git a/PKGBUILD b/PKGBUILD
index 2bc9ba9..634f026 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,18 +17,18 @@ b2sums=('0dd17b4fd358beb9842889168437443137445a5dba1f0a7e8669ae420d8efb927815c08
 
 build() {
   local prepare_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --disable-exec-static-tramp
     --disable-multi-os-directory
     --disable-static
     --enable-pax_emutramp
     --libdir=/usr/lib32
+    --includedir=/usr/include32
     --prefix=/usr
   )
 
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export LDFLAGS+=" -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
   export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
   cd $_name-$pkgver
@@ -39,7 +39,7 @@ build() {
 package() {
   cd $_name-$pkgver
   make DESTDIR="$pkgdir" install
-  rm -r "$pkgdir"/usr/{include,share}
+  rm -r "$pkgdir"/usr/share
   install -Dm 644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
 
