diff --git i/PKGBUILD w/PKGBUILD
index 7468d50..2c788a2 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -46,8 +46,9 @@ prepare() {
 
 build() {
   local configure_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --prefix=/usr
+    --includedir=/usr/include32
     --sysconfdir=/etc
     --libdir=/usr/lib32
     --program-prefix="eu-"
@@ -58,17 +59,20 @@ build() {
     --disable-static
   )
 
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export LDFLAGS+=" -m32"
-  export PKG_CONFIG=i686-pc-linux-gnu-pkg-config
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   # debugging information is required for test-suite
   CFLAGS+=" -g"
 
   cd $_name
   ./configure "${configure_options[@]}"
-  make
+
+  # -I/usr/include to -I/usr/include32 and -I/usr/include3232 to -I/usr/include32 in Makefile
+  sed -i 's|/usr/include|/usr/include32|g' **/Makefile
+  sed -i 's|/usr/include3232|/usr/include32|g' **/Makefile
+  make -j$(nproc)
 }
 
 check() {
@@ -82,7 +86,7 @@ package() {
   make DESTDIR="$pkgdir" install -C $_name
 
   # remove anything that would conflict with elfutils' packages
-  rm -rf "$pkgdir/"{etc,usr/{bin,include,share}}
+  rm -rf "$pkgdir/"{etc,usr/{bin,share}}
 }
 
 # vim:set sw=2 sts=-1 et:
