diff --git a/PKGBUILD b/PKGBUILD
index 80bd84b..0ca8ef7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,7 +36,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D avahi=disabled
     -D avb=disabled
     -D bluez5-codec-lc3=disabled
@@ -125,12 +125,13 @@ package_lib32-pipewire() {
     _pick lib usr/lib32/lib$_pwname.so*
     _pick lib usr/lib32/pkgconfig/lib{$_pwname,$_spaname}.pc
 
+    _pick jack usr/include32/jack
     _pick jack usr/lib32/libjack*
     _pick jack usr/lib32/pkgconfig/jack.pc
 
     _pick v4l2 usr/lib32/$_pwname/v4l2
 
-    rm -r usr/{bin,include,lib,share}
+    rm -r usr/{bin,lib,share}
   )
 
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 pipewire/COPYING
