diff --git a/PKGBUILD b/PKGBUILD
index 641796f..faff4c8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,14 +36,15 @@ build() {
     --enable-pcre2-16
     --enable-pcre2-32
     --libdir=/usr/lib32
+    --includedir=/usr/include32
     --prefix=/usr
   )
 
   cd $_name-$pkgver
 
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   ./configure "${configure_options[@]}"
   make
@@ -56,7 +57,7 @@ check() {
 package() {
   make DESTDIR="$pkgdir" install -C $_name-$pkgver
 
-  rm -rv "$pkgdir"/usr/{bin,share,include}
+  rm -rv "$pkgdir"/usr/{bin,share}
 
   install -vDm644 $_name-$pkgver/LICENCE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
