diff --git a/PKGBUILD b/PKGBUILD
index 711b929..7ad45ca 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -84,29 +84,33 @@ build() {
     -DCMAKE_INSTALL_DOCDIR=share/doc
     -DCMAKE_INSTALL_PREFIX=/usr
     -DCMAKE_SKIP_RPATH=ON
-    -DLLVM_BINUTILS_INCDIR=/usr/include
+    -DLLVM_BINUTILS_INCDIR=/usr/include32
+    -DCMAKE_INSTALL_INCLUDEDIR=/usr/include32
+    -DCMAKE_INSTALL_LIBDIR=/usr/lib32
+    -DCMAKE_INSTALL_BINDIR=/usr/armv7h-linux-gnueabihf/bin/
+    -DCMAKE_INSTALL_PREFIX=/usr/armv7h-linux-gnueabihf
     -DLLVM_BUILD_DOCS=OFF
     -DLLVM_BUILD_LLVM_DYLIB=ON
     -DLLVM_BUILD_TESTS=OFF
-    -DLLVM_DEFAULT_TARGET_TRIPLE="i686-pc-linux-gnu"
+    -DLLVM_DEFAULT_TARGET_TRIPLE="armv7h-linux-gnueabihf"
     -DLLVM_ENABLE_BINDINGS=OFF
     -DLLVM_ENABLE_CURL=OFF
     -DLLVM_ENABLE_FFI=ON
     -DLLVM_ENABLE_RTTI=ON
     -DLLVM_ENABLE_SPHINX=OFF
-    -DLLVM_HOST_TRIPLE=$CHOST
+    -DLLVM_TARGETS_TO_BUILD="AMDGPU;ARM"
+    -DLLVM_HOST_TRIPLE="armv7h-linux-gnueabihf"
     -DLLVM_INCLUDE_BENCHMARKS=OFF
     -DLLVM_INSTALL_GTEST=ON
-    -DLLVM_LIBDIR_SUFFIX=32
     -DLLVM_LINK_LLVM_DYLIB=ON
-    -DLLVM_TARGET_ARCH:STRING=i686
+    -DLLVM_TARGET_ARCH:STRING=arm
     -DLLVM_USE_PERF=ON
     -DPACKAGE_BUGREPORT=https://gitlab.archlinux.org/archlinux/packaging/packages/lib32-llvm/-/issues
   )
 
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cmake .. "${cmake_args[@]}"
   local distribution_components=$(_get_distribution_components | paste -sd\;)
@@ -124,15 +128,16 @@ package_lib32-llvm() {
   cd llvm-$pkgver.src/build
 
   DESTDIR="$pkgdir" ninja install-distribution
-
+  install -d "$pkgdir/usr/lib32"
+  cp -r "$pkgdir"/usr/armv7h-linux-gnueabihf/lib/* "$pkgdir"/usr/lib32/
+  rm -rf "$pkgdir"/usr/armv7h-linux-gnueabihf/lib/
   # The runtime library goes into lib32-llvm-libs
   mv "$pkgdir"/usr/lib32/lib{LLVM,LTO,Remarks}*.so* "$srcdir"
   mv -f "$pkgdir"/usr/lib32/LLVMgold.so "$srcdir"
 
 
-  rm -rf "$pkgdir"/usr/{include,share/{doc,man,llvm,opt-viewer}}
-  find "$pkgdir/usr/bin" -mindepth 1 -not -name llvm-config -delete
-  mv "$pkgdir"/usr/bin/llvm-config{,32}
+  rm -rf "$pkgdir"/usr/armv7h-linux-gnueabihf/share
+  # mv "$pkgdir"/usr/bin/llvm-config{,32}
 
   install -Dm644 ../LICENSE.TXT "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
