diff --git a/PKGBUILD b/PKGBUILD
index fdeaaf0..b4fe4c6 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,7 +42,8 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --libdir /usr/lib32
+    --includedir /usr/include32
     -D builtin_loaders=all
     -D gtk_doc=false
     -D installed_tests=false
@@ -51,6 +52,10 @@ build() {
     -D others=enabled
   )
 
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
+
   arch-meson gdk-pixbuf build "${meson_options[@]}"
   meson compile -C build
 }
@@ -65,7 +70,7 @@ package() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -rf "$pkgdir"/usr/{include,lib,share}
+  rm -rf "$pkgdir"/usr/share
   find "$pkgdir/usr/bin" -type f -not -name gdk-pixbuf-query-loaders -delete
   mv "$pkgdir"/usr/bin/gdk-pixbuf-query-loaders{,-32}
 
