diff --git a/PKGBUILD b/PKGBUILD
index 9e16fc9..190f4bd 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,20 +46,19 @@ prepare() {
 
 build() {
   local configure_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --disable-debuginfod
     --disable-static
     --libdir=/usr/lib32
     --libexecdir=/usr/lib32
-    --includedir=/usr/include/${_name}32
+    --includedir=/usr/include32
     --prefix=/usr
     --program-suffix="-32"
     --sysconfdir=/etc
   )
 
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export LDFLAGS+=" -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
   export PKG_CONFIG_PATH="/usr/lib32/pkgconfig"
   # fat-lto-objects is required for non-mangled .a files in libelf
   CFLAGS+=" -ffat-lto-objects"
@@ -68,7 +67,7 @@ build() {
 
   cd $_name-$pkgver
   ./configure "${configure_options[@]}"
-  make
+  make -j$(nproc)
 }
 
 check() {
@@ -82,5 +81,5 @@ package() {
   make DESTDIR="$pkgdir" install -C $_name-$pkgver/libelf
   make DESTDIR="$pkgdir" install -C $_name-$pkgver/config
   # remove anything that would conflict with elfutils and debuginfod
-  rm -rf "$pkgdir/"{etc,usr/{bin,include,share}}
+  rm -rf "$pkgdir/"{etc,usr/{bin,share}}
 }
