diff --git a/PKGBUILD b/PKGBUILD
index bee259c..8288244 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -48,21 +48,21 @@ build() {
     -D CMAKE_INSTALL_SYSCONFDIR=/etc
     -D CMAKE_POSITION_INDEPENDENT_CODE=ON
     -D CMAKE_SKIP_RPATH=ON
-    -D LLVM_CONFIG=llvm-config32
+    -D LLVM_CONFIG="/usr/armv7h-linux-gnueabihf/llvm-config"
     -D LLVM_EXTERNAL_LIT=/usr/bin/lit
-    -D LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=/usr/include/spirv
+    -D LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=/usr/include32/spirv/
     -D LLVM_LIBDIR_SUFFIX=32
     -D LLVM_SPIRV_ENABLE_LIBSPIRV_DIS=ON
     -D LLVM_SPIRV_INCLUDE_TESTS=ON
     -W no-dev
   )
 
-  export ASFLAGS+=" --32"
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cmake -S SPIRV-LLVM-Translator -B build -G Ninja "${cmake_options[@]}"
+  sed -i 's|/usr/lib/libLLVM.so|/usr/lib32/libLLVM.so|g' build/build.ninja
   cmake --build build
 }
 
@@ -73,7 +73,9 @@ check() {
 
 package() {
   DESTDIR="$pkgdir" cmake --install build
-  rm -r "$pkgdir"/usr/{bin,include}
+  rm -r "$pkgdir"/usr/bin
+  mv ${pkgdir}/usr/include ${pkgdir}/usr/include32
+  sed -i 's|${prefix}/include|${prefix}/include32|' ${pkgdir}/usr/lib32/pkgconfig/*.pc
 
   install -Dm644 SPIRV-LLVM-Translator/LICENSE.TXT \
     -t "$pkgdir/usr/share/licenses/$pkgname"
