diff --git a/PKGBUILD b/PKGBUILD
index e6d1875..2b737a3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,7 +39,8 @@ build() {
     --sysconfdir=/etc
     --localstatedir=/var
     --libdir=/usr/lib32
-    --libexecdir=/usr/lib/dbus-1.0
+    --includedir=/usr/include32
+    --libexecdir=/usr/lib32/dbus-1.0
     --runstatedir=/run
     --with-console-auth-dir=/run/console/
     --with-dbus-user=dbus
@@ -57,9 +58,9 @@ build() {
     --without-x
   )
 
-  export CC="gcc -m32 -mstackrealign"
-  export CXX="g++ -m32 -mstackrealign"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cd dbus-$pkgver
   ./configure "${configure_options[@]}"
@@ -85,7 +86,7 @@ package_lib32-dbus() {
   DESTDIR="$pkgdir" make install
 
   rm -r "$pkgdir"/{etc,var}
-  rm -r "$pkgdir"/usr/{bin,include,lib,share}
+  rm -r "$pkgdir"/usr/{bin,lib,share}
 
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 COPYING
 }
