diff --git a/PKGBUILD b/PKGBUILD
index bb6a532..a2cdb90 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -39,7 +39,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D android=disabled
     -D builtin_loaders=all
     -D documentation=false
@@ -56,6 +56,10 @@ build() {
     -D tiff=enabled
   )
 
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
+
   arch-meson gdk-pixbuf build "${meson_options[@]}"
   meson compile -C build
 }
@@ -69,7 +73,7 @@ package() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -rf "$pkgdir"/usr/{include,lib,share}
+  rm -rf "$pkgdir"/usr/share
   find "$pkgdir/usr/bin" -type f -not -name gdk-pixbuf-query-loaders -delete
   mv "$pkgdir"/usr/bin/gdk-pixbuf-query-loaders{,-32}
 
