diff --git i/PKGBUILD w/PKGBUILD
index a5216e9..732e24b 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -36,9 +36,11 @@ validpgpkeys=(
 )
 
 build() {
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export LDFLAGS+=" -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export CFLAGS="-I/usr/armv7h-linux-gnueabihf/sys-root/usr/include"
+  export CXXFLAGS="-I/usr/armv7h-linux-gnueabihf/sys-root/usr/include"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
   export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
   local cmake_options=(
@@ -46,17 +48,27 @@ build() {
     -D BUILD_SHARED_LIBS=ON
     -D CMAKE_BUILD_TYPE=None
     -D CMAKE_INSTALL_LIBDIR=lib32
+    -D CMAKE_SYSTEM_INCLUDE_PATH=/usr/include32
+    -D CMAKE_SYSROOT=/usr/armv7h-linux-gnueabihf/sys-root
+    -D CMAKE_INSTALL_INCLUDEDIR=include32
     -D CMAKE_INSTALL_PREFIX=/usr
+    -D CMAKE_POLICY_VERSION_MINIMUM=3.5
+    -D ENABLE_EXTERNAL_LIBS=ON
     -D ENABLE_MPEG=OFF
     -S $_name-$pkgver
     -W no-dev
   )
+
   local cmake_test_options=(
     -B build-test
     -D CMAKE_BUILD_TYPE=None
     -D CMAKE_INSTALL_LIBDIR=lib32
+    -D CMAKE_SYSTEM_INCLUDE_PATH=/usr/include32
+    -D CMAKE_SYSROOT=/usr/armv7h-linux-gnueabihf/sys-root
+    -D CMAKE_INSTALL_INCLUDEDIR=include32
     -D CMAKE_INSTALL_PREFIX=/usr
     -D ENABLE_MPEG=OFF
+    -D CMAKE_POLICY_VERSION_MINIMUM=3.5
     -S $_name-$pkgver
     -W no-dev
   )
@@ -82,5 +94,5 @@ package() {
   install -vDm 644 $_name-$pkgver/{AUTHORS,ChangeLog,README} -t "$pkgdir/usr/share/doc/$pkgname"
 
   # remove everything that is provided by libsndfile
-  rm -rf "$pkgdir/usr/"{bin,include,share/{man,doc/$_name}}
+  rm -rf "$pkgdir/usr/"{bin,share/{man,doc/$_name}}
 }
