diff --git a/PKGBUILD b/PKGBUILD
index c7dc233..1073e62 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,9 +17,9 @@ source=("https://github.com/thom311/libnl/releases/download/libnl${pkgver//./_}/
 sha512sums=('69ecec6e792fc7b9c443fff8742cf45782af5c5c4664687440942eaeb616ba7b4ed2b606e33c5d86e44e6b49a9c79a1fed4b7c77781a059e13cf6a844d94530e')
 
 build() {
-  export CC='gcc -m32'
-  export CXX='g++ -m32'
-  export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cd "${_pkgbase}-${pkgver}"
   ./configure \
@@ -27,12 +27,13 @@ build() {
     --sysconfdir=/etc \
     --sbindir=/usr/bin \
     --disable-static \
-    --libdir=/usr/lib32
+    --libdir=/usr/lib32 \
+    --includedir /usr/include32
   make
 }
 
 package() {
   cd "${_pkgbase}-${pkgver}"
   make DESTDIR="${pkgdir}" install
-  rm -rf "${pkgdir}"/{etc,usr/{bin,include,share}}
+  rm -rf "${pkgdir}"/{etc,usr/{bin,share}}
 }
