diff --git a/PKGBUILD b/PKGBUILD
index b114315..981d351 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,13 +14,14 @@ sha256sums=('0e982409a9c3fc82ee06e08205b1355e5c6aa4c36bca58146ef399621b0ce5ab')
 build() {
   cd libvorbis-${pkgver}
 
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   ./configure \
     --prefix='/usr' \
     --libdir='/usr/lib32' \
+    --includedir='/usr/include32' \
     --disable-static
   make
 }
@@ -36,7 +37,7 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 
-  rm -rf "${pkgdir}"/usr/{include,share,bin}
+  rm -rf "${pkgdir}"/usr/{share,bin}
   install -d "$pkgdir/usr/share/licenses"
   ln -s libvorbis "$pkgdir/usr/share/licenses/$pkgname"
 }
