diff --git a/PKGBUILD b/PKGBUILD
index 0f57219..8c861ab 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,7 +50,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D chafa=disabled
     -D cpp_std=c++17
     -D docs=disabled
@@ -62,6 +62,7 @@ build() {
   CXXFLAGS="${CXXFLAGS/-fexceptions/}"
 
   arch-meson harfbuzz build "${meson_options[@]}"
+  sed -i 's|-I/usr/include |-I/usr/include32 |' build/{build.ninja,compile_commands.json}
   meson compile -C build
 }
 
@@ -91,7 +92,7 @@ package_lib32-harfbuzz() {
   meson install -C build --destdir "$pkgdir"
 
   ( cd "$pkgdir"
-    rm -r usr/{bin,include}
+    rm -r usr/bin
 
     _pick hb-cairo usr/lib32/libharfbuzz-cairo*
     _pick hb-cairo usr/lib32/pkgconfig/harfbuzz-cairo.pc
