diff --git a/PKGBUILD b/PKGBUILD
index 0eb998b..48f2582 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,26 +23,30 @@ pkgver() {
 }
 
 build() {
-  export PKG_CONFIG=i686-pc-linux-gnu-pkg-config
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cmake -B build -S ${_srcname} \
+    -GNinja \
     -DBUILD_SHARED_LIBS=ON \
     -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_CXX_FLAGS:STRING=-m32 \
-    -DCMAKE_C_FLAGS:STRING=-m32 \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
     -DCMAKE_SKIP_RPATH=ON \
-    -DLLVM_CONFIG="/usr/bin/llvm-config32" \
+    -DLLVM_CONFIG="/usr/armv7h-linux-gnueabihf/llvm-config" \
     -DLLVM_INCLUDE_TESTS=OFF \
-    -DLLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=/usr/include/spirv/ \
+    -DLLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR=/usr/include32/spirv/ \
     -DLLVM_LIBDIR_SUFFIX=32 \
     -Wno-dev
-  make -C build
+  sed -i 's|/usr/lib/libLLVM.so|/usr/lib32/libLLVM.so|g' build/build.ninja
+  ninja -C build
 }
 
 package() {
-  make -C build DESTDIR="${pkgdir}" install
-  rm -r "${pkgdir}"/usr/{bin,include}
+  DESTDIR="${pkgdir}" cmake --install build
+  rm -r "${pkgdir}"/usr/bin
+  mv ${pkgdir}/usr/include ${pkgdir}/usr/include32
+  sed -i 's|${prefix}/include|${prefix}/include32|' ${pkgdir}/usr/lib32/pkgconfig/spirv-llvm-translator.pc
   install -Dm644 ${_srcname}/LICENSE.TXT -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
