diff --git a/PKGBUILD b/PKGBUILD
index 564c93c..a50f2b8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -60,7 +60,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D documentation=false
     -D dtrace=disabled
     -D glib_debug=disabled
@@ -76,8 +76,8 @@ build() {
   CXXFLAGS+=" -g3"
 
   # Avoid crashing some old binaries
-  CFLAGS+=" -mstackrealign"
-  CXXFLAGS+=" -mstackrealign"
+  #CFLAGS+=" -mstackrealign"
+  #CXXFLAGS+=" -mstackrealign"
 
   arch-meson glib build "${meson_options[@]}"
   meson compile -C build
@@ -96,7 +96,7 @@ package_lib32-glib2() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir"/usr/{lib,share,include}
+  rm -r "$pkgdir"/usr/{lib,share}
   find "$pkgdir/usr/bin" -type f -not -name gio-querymodules -printf 'Removing %P\n' -delete
   mv "$pkgdir"/usr/bin/gio-querymodules{,-32}
 
