diff --git i/PKGBUILD w/PKGBUILD
index 9491888..8013d38 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -42,8 +42,13 @@ prepare() {
 }
 
 build() {
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
+
   local meson_options=(
-    --cross-file lib32
+    --libdir=/usr/lib32
+    --includedir=/usr/include32
     -D bash_completion=false
     -D daemon=false
     -D docs=false
@@ -62,9 +67,10 @@ check() {
 
 package() {
   meson install -C build --destdir "$pkgdir"
-  rm -r "${pkgdir}"/usr/{bin,include,lib,share}
+  rm -r "${pkgdir}"/usr/{bin,share}
   rm "${pkgdir}"/usr/lib32/libcolor{dprivate,hug}*
   rm "${pkgdir}"/usr/lib32/pkgconfig/colorhug.pc
+  mv "${pkgdir}"/usr/lib/* "${pkgdir}"/usr/lib32
 }
 
 # vim:set sw=2 sts=-1 et:
