diff --git a/PKGBUILD b/PKGBUILD
index 3da3d7c..79bcba2 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,12 +14,12 @@ source=(${url}/releases/individual/lib/libXcomposite-${pkgver}.tar.xz)
 sha512sums=('f35a0da8f9f22fd788eee01209557fe3a1b1abdc9744f76330ba42b21b6e3af5464ca4451c33415856417c30da5aa5624e3707980c644327249bfeda5a62a59b')
 
 build() {
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export PKG_CONFIG_PATH="/usr/lib32/pkgconfig"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cd "${srcdir}/libXcomposite-${pkgver}"
-  ./configure --prefix=/usr --sysconfdir=/etc --disable-static --libdir=/usr/lib32
+  ./configure --prefix=/usr --sysconfdir=/etc --disable-static --libdir=/usr/lib32 --includedir=/usr/include32
   make
 }
 
@@ -28,7 +28,7 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 
-  rm -rf "${pkgdir}"/usr/{include,share,bin}
+  rm -rf "${pkgdir}"/usr/{share,bin}
   mkdir -p "$pkgdir/usr/share/licenses"
   ln -s $_pkgbasename "$pkgdir/usr/share/licenses/$pkgname"
 }
