diff --git a/PKGBUILD b/PKGBUILD
index b6c879a..e804090 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,7 @@ makedepends=(
   git
   lib32-libx11
   meson
+  lib32-valgrind
   valgrind
 )
 optdepends=('lib32-alsa-plugins: ALSA support')
@@ -49,7 +50,8 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --libdir /usr/lib32
+    --includedir /usr/include32
     -D avahi=disabled
     -D bashcompletiondir=no
     -D bluez5-gstreamer=disabled
@@ -76,7 +78,11 @@ build() {
     -D zshcompletiondir=no
   )
 
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
   arch-meson pulseaudio build "${meson_options[@]}"
+  sed -i 's|-I/usr/include |-I/usr/include32 |' $srcdir/build/{compile_commands.json,build.ninja}
   meson compile -C build
 }
 
@@ -86,7 +92,7 @@ check() {
 
 package_lib32-libpulse() {
   meson install -C build --destdir "$pkgdir"
-  rm -r "$pkgdir/etc" "$pkgdir"/usr/{bin,include,share}
+  rm -r "$pkgdir/etc" "$pkgdir"/usr/{bin,share}
 }
 
 # vim:set sw=2 et:
