diff --git a/PKGBUILD b/PKGBUILD
index 68d923c..e1648fe 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,25 +7,18 @@
 pkgbase=lib32-mesa
 pkgname=(
   lib32-mesa
-  lib32-opencl-mesa
-  lib32-vulkan-asahi
   lib32-vulkan-broadcom
-  lib32-vulkan-dzn
   lib32-vulkan-freedreno
-  lib32-vulkan-gfxstream
-  lib32-vulkan-intel
-  lib32-vulkan-nouveau
   lib32-vulkan-panfrost
-  lib32-vulkan-powervr
   lib32-vulkan-radeon
   lib32-vulkan-swrast
   lib32-vulkan-virtio
   lib32-vulkan-mesa-implicit-layers
   lib32-vulkan-mesa-layers
 )
-pkgver=25.3.3
+pkgver=25.3.4
 _pkgver=${pkgver/[a-z]/-&}
-pkgrel=3
+pkgrel=1
 epoch=1
 pkgdesc="Open-source OpenGL drivers - 32-bit"
 url="https://www.mesa3d.org/"
@@ -53,7 +46,6 @@ makedepends=(
   lib32-llvm
   lib32-llvm-libs
   lib32-lm_sensors
-  lib32-rust-libs
   lib32-spirv-llvm-translator
   lib32-spirv-tools
   lib32-systemd
@@ -76,9 +68,10 @@ makedepends=(
   python-ply
   python-pycparser
   python-yaml
-  rust-bindgen
   wayland-protocols
-  xorgproto
+  lib32-directx-headers
+  lib32-libomxil-bellagio
+  lib32-xorgproto
 )
 options=(
   # GCC 14 LTO causes segfault in LLVM under si_llvm_optimize_module
@@ -226,14 +219,13 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=all
+    -D gallium-drivers=r300,r600,radeonsi,freedreno,nouveau,softpipe,llvmpipe,virgl,d3d12,etnaviv,lima,panfrost,svga,v3d,vc4
     -D gallium-extra-hud=true
     -D gallium-mediafoundation=disabled
-    -D gallium-rusticl-enable-drivers=asahi,freedreno,radeonsi
-    -D gallium-rusticl=true
+    -D gallium-rusticl=false
     -D gles1=disabled
     -D html-docs=disabled
     -D intel-rt=disabled
@@ -242,8 +234,8 @@ build() {
     -D sysprof=false
     -D valgrind=disabled
     -D video-codecs=all
-    -D vulkan-drivers=all
-    -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,anti-lag,vram-report-limit
+    -D vulkan-drivers=amd,swrast,broadcom,panfrost,virtio,freedreno
+    -D vulkan-layers=device-select,overlay,screenshot,anti-lag,vram-report-limit
   )
 
   # Build only minimal debug info to reduce size
@@ -312,36 +306,15 @@ package_lib32-mesa() {
 
     cd "$pkgdir"
 
-    _pick opencl $libdir/libRusticlOpenCL*
-
-    _pick vkasahi $icddir/asahi_icd.*.json
-    _pick vkasahi $libdir/libvulkan_asahi.so
-
     _pick vkbrcom $icddir/broadcom_icd.*.json
     _pick vkbrcom $libdir/libvulkan_broadcom.so
 
-    _pick vkd3d12 $icddir/dzn_icd.*.json
-    _pick vkd3d12 $libdir/libvulkan_dzn.so
-    _pick vkd3d12 $libdir/libspirv_to_dxil.*
-
     _pick vkfdreno $icddir/freedreno_icd.*.json
     _pick vkfdreno $libdir/libvulkan_freedreno.so
 
-    _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
-    _pick vkgfxstr $libdir/libvulkan_gfxstream.so
-
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
-
-    _pick vknvidia $icddir/nouveau_icd.*.json
-    _pick vknvidia $libdir/libvulkan_nouveau.so
-
     _pick vkpfrost $icddir/panfrost_icd.*.json
     _pick vkpfrost $libdir/libvulkan_panfrost.so
 
-    _pick vkpowrvr $icddir/powervr_mesa_icd.*.json
-    _pick vkpowrvr $libdir/libvulkan_powervr_mesa.so
-
     _pick vkradeon $icddir/radeon_icd.*.json
     _pick vkradeon $libdir/libvulkan_radeon.so
 
@@ -355,8 +328,7 @@ package_lib32-mesa() {
     _pick vkdevice $libdir/libVkLayer_MESA_device_select.so
 
     _pick vklayer $libdir/libVkLayer_*.so
-
-    rm -rv etc usr/{bin,include,share}
+    rm -frv etc usr/{bin,share}
 
     # indirect rendering
     ln -sr $libdir/libGLX_{mesa,indirect}.so.0
