diff --git a/PKGBUILD b/PKGBUILD
index 2e3ac20..bb04359 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -38,7 +38,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D enable-bash-completion=false
     -D enable-tools=false
   )
@@ -74,7 +74,6 @@ package_lib32-libxkbcommon() {
   )
 
   meson install -C build --destdir "$pkgdir"
-  rm -r "$pkgdir/usr/include"
 
   _pick x11 "$pkgdir"/usr/lib32/*x11*
   _pick x11 "$pkgdir"/usr/lib32/pkgconfig/*x11*
