diff --git a/PKGBUILD b/PKGBUILD
index 668f0eb..446067b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -53,16 +53,17 @@ prepare() {
 build() {
    cd "${srcdir}/${_pkgbasename}-${pkgver}/src"
 
-   export CC="gcc -m32"
-   export CXX="g++ -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
    export PKG_CONFIG_PATH="/usr/lib32/pkgconfig"
 
    export CFLAGS+=" -fPIC -fno-strict-aliasing -fstack-protector-all"
-   export CPPFLAGS+=" -I/usr/include/et"
+   export CPPFLAGS+=" -I/usr/include32/et"
    ./configure --prefix=/usr \
                --sysconfdir=/etc \
                --localstatedir=/var/lib \
                --libdir=/usr/lib32 \
+               --includedir=/usr/include32 \
                --enable-shared \
                --with-system-et \
                --with-system-ss \
@@ -86,7 +87,7 @@ package() {
    cd ${_pkgbasename}-${pkgver}/src
    make DESTDIR="${pkgdir}" install
 
-   rm -rf "${pkgdir}"/usr/{include,share,bin,sbin}
+   rm -rf "${pkgdir}"/usr/{share,bin,sbin}
    install -d "${pkgdir}/usr/share/licenses"
    ln -s ${_pkgbasename} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
