diff --git i/PKGBUILD w/PKGBUILD
index 325174a..f842ea2 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -3,7 +3,7 @@
 pkgname=lib32-libarchive
 _name="${pkgname#lib32-}"
 pkgver=3.8.2
-pkgrel=1
+pkgrel=2
 pkgdesc="Multi-format archive and compression library - 32bit"
 arch=(x86_64)
 url="https://libarchive.org"
@@ -66,7 +66,7 @@ prepare() {
 
 build() {
   local configure_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --disable-bsdcat
     --disable-bsdcpio
     --disable-bsdtar
@@ -74,15 +74,13 @@ build() {
     --disable-static
     --libdir=/usr/lib32
     --libexecdir=/usr/lib32
-    --includedir=/usr/include/${_name}32
+    --includedir=/usr/include32
     --prefix=/usr
     --program-suffix="-32"
     --without-nettle
   )
-
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export LDFLAGS+=" -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
   export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
   cd $_name
@@ -95,6 +93,6 @@ check() {
 }
 
 package() {
-  make DESTDIR="$pkgdir/" install-libLTLIBRARIES install-pkgconfigDATA -C $_name
+  make DESTDIR="$pkgdir/" install-libLTLIBRARIES install-pkgconfigDATA install-includeHEADERS -C $_name
   install -vDm 644 ./*.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
