diff --git a/PKGBUILD b/PKGBUILD
index 4315031..c9dafd9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,12 +20,13 @@ sha256sums=('5f2bdbad629707aa7d85c623f994aa8a1d2dec55a73de5205bac0bf6058a2f7c'
 build() {
   cd "${srcdir}"/acl-${pkgver}
 
-  export CC="gcc -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
   export PKG_CONFIG_PATH="/usr/lib32/pkgconfig"
 
   ./configure \
     --prefix=/usr \
     --libdir=/usr/lib32 \
+    --includedir=/usr/include32 \
     --libexecdir=/usr/lib32
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make 
@@ -36,5 +37,5 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 
-  rm -rf "${pkgdir}"/usr/{bin,include,share}
+  rm -rf "${pkgdir}"/usr/{bin,share}
 }
