diff --git a/PKGBUILD b/PKGBUILD
index c23b960..1838689 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -40,7 +40,7 @@ prepare() {
 
 build() {
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D apparmor=disabled
     -D dbus_user=dbus
     -D doxygen_docs=disabled
@@ -78,8 +78,6 @@ package_lib32-dbus() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir/usr/include"
-
   mkdir -p "$pkgdir/usr/share/licenses"
   ln -s dbus "$pkgdir/usr/share/licenses/$pkgname"
 }
