diff --git a/PKGBUILD b/PKGBUILD
index 6063062..ccdbe34 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,15 +26,16 @@ prepare() {
 }
 
 build() {
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export PKG_CONFIG="i686-pc-linux-gnu-pkg-config"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cd libpng
 
   ./configure \
     --prefix=/usr \
     --libdir=/usr/lib32 \
+    --includedir=/usr/include32 \
     --program-suffix=-32 \
     --disable-static
   make
@@ -45,7 +46,7 @@ package() {
 
   make DESTDIR="${pkgdir}" install
 
-  rm -rf "${pkgdir}"/usr/{include,share}
+  rm -rf "${pkgdir}"/usr/share
 
   rm "$pkgdir/usr/bin/libpng-config"
   ln -s "libpng${_libversion}-config-32" "$pkgdir/usr/bin/libpng-config-32"
