diff --git a/PKGBUILD b/PKGBUILD
index 95d7cf1..37cd54c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,15 +8,13 @@ pkgbase=lib32-mesa
 pkgname=(
   lib32-mesa
   lib32-opencl-clover-mesa
-  lib32-opencl-rusticl-mesa
-  lib32-vulkan-dzn
-  lib32-vulkan-gfxstream
-  lib32-vulkan-intel
-  lib32-vulkan-nouveau
   lib32-vulkan-radeon
   lib32-vulkan-swrast
   lib32-vulkan-virtio
   lib32-vulkan-mesa-layers
+  lib32-vulkan-broadcom
+  lib32-vulkan-panfrost
+  lib32-vulkan-freedreno
 )
 pkgver=25.0.4
 pkgrel=1
@@ -47,7 +45,6 @@ makedepends=(
   lib32-llvm
   lib32-llvm-libs
   lib32-lm_sensors
-  lib32-rust-libs
   lib32-spirv-llvm-translator
   lib32-spirv-tools
   lib32-systemd
@@ -69,9 +66,10 @@ makedepends=(
   python-packaging
   python-ply
   python-yaml
-  rust-bindgen
   wayland-protocols
-  xorgproto
+  lib32-directx-headers
+  lib32-libomxil-bellagio
+  lib32-xorgproto
 )
 options=(
   # GCC 14 LTO causes segfault in LLVM under si_llvm_optimize_module
@@ -171,15 +169,17 @@ prepare() {
 }
 
 build() {
+  # shellcheck disable=SC2054
   local meson_options=(
-    --cross-file lib32
+    --cross-file arm-lib32
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,freedreno,nouveau,softpipe,llvmpipe,virgl,d3d12,etnaviv,lima,panfrost,svga,v3d,vc4
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-opencl=icd
-    -D gallium-rusticl=true
+    -D gallium-rusticl=false
+    -D gallium-xa=disabled
     -D gles1=disabled
     -D html-docs=disabled
     -D libunwind=disabled
@@ -187,19 +187,16 @@ build() {
     -D osmesa=true
     -D valgrind=disabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,gfxstream,intel,intel_hasvk,nouveau,swrast,virtio,microsoft-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,vram-report-limit
+    -D vulkan-drivers=amd,swrast,broadcom,panfrost,virtio,freedreno
+    -D vulkan-layers=device-select,overlay
   )
 
-  export BINDGEN_EXTRA_CLANG_ARGS="-m32"
-
   # Build only minimal debug info to reduce size
   CFLAGS+=" -g1"
   CXXFLAGS+=" -g1"
 
   # Inject subproject packages
   export MESON_PACKAGE_CACHE_DIR="$srcdir"
-
   arch-meson mesa-$pkgver build "${meson_options[@]}"
   meson compile -C build
 }
@@ -266,21 +263,6 @@ package_lib32-mesa() {
     _pick clover $libdir/gallium-pipe
     _pick clover $libdir/libMesaOpenCL*
 
-    _pick clrust $libdir/libRusticlOpenCL*
-
-    _pick vkdzn $icddir/dzn_icd.*.json
-    _pick vkdzn $libdir/libvulkan_dzn.so
-    _pick vkdzn $libdir/libspirv_to_dxil.*
-
-    _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
-    _pick vkgfxstr $libdir/libvulkan_gfxstream.so
-
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
-
-    _pick vknvidia $icddir/nouveau_icd.*.json
-    _pick vknvidia $libdir/libvulkan_nouveau.so
-
     _pick vkradeon $icddir/radeon_icd.*.json
     _pick vkradeon $libdir/libvulkan_radeon.so
 
@@ -292,7 +274,16 @@ package_lib32-mesa() {
 
     _pick vklayer $libdir/libVkLayer_*.so
 
-    rm -rv etc usr/{bin,include,share}
+    _pick vkbroadcom $icddir/broadcom_icd*.json
+    _pick vkbroadcom $libdir/libvulkan_broadcom.so
+
+    _pick vkpanfrost $icddir/panfrost_icd*.json
+    _pick vkpanfrost $libdir/libvulkan_panfrost.so
+
+    _pick vkfreedreno $icddir/freedreno_icd*.json
+    _pick vkfreedreno $libdir/libvulkan_freedreno.so
+
+    rm -rv etc usr/{bin,share}
 
     # indirect rendering
     ln -sr $libdir/libGLX_{mesa,indirect}.so.0
@@ -572,3 +563,50 @@ package_lib32-vulkan-mesa-layers() {
 }
 
 # vim:set sw=2 sts=-1 et:
+package_lib32-vulkan-broadcom() {
+  pkgdesc="Broadcom's Vulkan mesa driver - 32-bit"
+  depends=(
+    lib32-wayland
+    lib32-libx11
+    lib32-libxshmfence
+    lib32-libdrm
+  )
+  optdepends=("lib32-vulkan-mesa-layers: additional vulkan layers")
+  provides=(lib32-vulkan-driver)
+
+  mv vkbroadcom/* "$pkgdir"
+
+  install -Dm644 mesa-$pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
+}
+
+package_lib32-vulkan-panfrost() {
+  pkgdesc="Panfrost Vulkan mesa driver - 32-bit"
+  depends=(
+    lib32-wayland
+    lib32-libx11
+    lib32-libxshmfence
+    lib32-libdrm
+  )
+  optdepends=("lib32-vulkan-mesa-layers: additional vulkan layers")
+  provides=(lib32-vulkan-driver)
+
+  mv vkpanfrost/* "$pkgdir"
+
+  install -Dm644 mesa-$pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
+}
+
+package_lib32-vulkan-freedreno() {
+  pkgdesc="Freedreno Vulkan mesa driver - 32-bit"
+  depends=(
+    lib32-wayland
+    lib32-libx11
+    lib32-libxshmfence
+    lib32-libdrm
+  )
+  optdepends=("lib32-vulkan-mesa-layers: additional vulkan layers")
+  provides=(lib32-vulkan-driver)
+
+  mv vkfreedreno/* "$pkgdir"
+
+  install -Dm644 mesa-$pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
+}
