diff --git i/PKGBUILD w/PKGBUILD
index 3ac85e7..3e714ac 100644
--- i/PKGBUILD
+++ w/PKGBUILD
@@ -13,8 +13,8 @@ license=(MIT)
 depends=(lib32-glibc $_name=$pkgver)
 provides=(libffi.so)
 source=($_name-$pkgver.tar.gz::$_url/archive/refs/tags/v$pkgver.tar.gz)
-sha256sums=('dd19253d3007f366319a51d248a40c9e5fcace4498cbea990b566291844e4e30')
-b2sums=('4aa7ecdbb7abc862f876cc0b0484f944fe6c49633071d36a964ad94f06d02ee08c76dceed5d6897da02c357515033f084a9a4da3a22bdd559316a5dca736332d')
+sha256sums=('SKIP')
+b2sums=('SKIP')
 
 prepare() {
   cd $_name-$pkgver
@@ -24,18 +24,18 @@ prepare() {
 
 build() {
   local prepare_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --disable-exec-static-tramp
     --disable-multi-os-directory
     --disable-static
     --enable-pax_emutramp
     --libdir=/usr/lib32
+    --includedir=/usr/include32
     --prefix=/usr
   )
 
-  export CFLAGS+=" -m32"
-  export CXXFLAGS+=" -m32"
-  export LDFLAGS+=" -m32"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
   export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
   cd $_name-$pkgver
@@ -46,7 +46,7 @@ build() {
 package() {
   cd $_name-$pkgver
   make DESTDIR="$pkgdir" install
-  rm -r "$pkgdir"/usr/{include,share}
+  rm -r "$pkgdir"/usr/share
   install -Dm 644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
 
