diff --git a/PKGBUILD b/PKGBUILD
index fc31ab0..64fd71b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,10 +27,11 @@ prepare() {
 }
 
 build() {
-  export CC='gcc -m32'
-  export PKG_CONFIG=i686-pc-linux-gnu-pkg-config
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
-  arch-meson libepoxy build --libdir=/usr/lib32
+  arch-meson libepoxy build --libdir=/usr/lib32 --includedir /usr/include32
   meson compile -C build
 }
 
@@ -44,8 +45,6 @@ package() {
 
   meson install -C build --destdir "$pkgdir"
 
-  rm -r "$pkgdir/usr/include"
-
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 libepoxy/COPYING
 }
 
