diff --git a/PKGBUILD b/PKGBUILD
index 7eeeb28..822bc7c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,20 +46,20 @@ prepare() {
 
 build() {
   local configure_options=(
-    --build=i686-pc-linux-gnu
+    --build=armv7h-linux-gnueabihf
     --disable-debuginfod
     --disable-static
     --libdir=/usr/lib32
     --libexecdir=/usr/lib32
     --prefix=/usr
+    --includedir=/usr/include32
     --program-suffix="-32"
     --sysconfdir=/etc
   )
 
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export LDFLAGS+=" -m32"
-  export PKG_CONFIG=i686-pc-linux-gnu-pkg-config
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
   # fat-lto-objects is required for non-mangled .a files in libelf
   CFLAGS+=" -ffat-lto-objects"
   # debugging information is required for test-suite
@@ -67,7 +67,10 @@ build() {
 
   cd $_name-$pkgver
   ./configure "${configure_options[@]}"
-  make
+  # -I/usr/include to -I/usr/include32 and -I/usr/include3232 to -I/usr/include32 in Makefile
+  sed -i 's|/usr/include|/usr/include32|g' **/Makefile
+  sed -i 's|/usr/include3232|/usr/include32|g' **/Makefile
+  make -j$(nproc)
 }
 
 check() {
@@ -84,5 +87,5 @@ package() {
   make DESTDIR="$pkgdir" install -C $_name-$pkgver/libelf
   make DESTDIR="$pkgdir" install -C $_name-$pkgver/config
   # remove anything that would conflict with elfutils and debuginfod
-  rm -rf "$pkgdir/"{etc,usr/{bin,include,share}}
+  rm -rf "$pkgdir/"{etc,usr/{bin,share}}
 }
