diff --git a/PKGBUILD b/PKGBUILD
index 8c23cbc..b763212 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,17 +13,16 @@ source=("https://gitlab.freedesktop.org/libdecor/libdecor/-/archive/${pkgver}/li
 sha512sums=('ddaaa75f1dc983e6d59e168896f8044cf6ac761675db6af410c6d2186bdc66942c98f36eb7a6e2836ce02dbb98189a80db30b7cef3dbaf49c7aae47c3eee22cc')
 
 build() {
-  export CC='gcc -m32'
-  export CXX='g++ -m32'
-  export PKG_CONFIG=i686-pc-linux-gnu-pkg-config
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   meson -Ddemo=false build libdecor-$pkgver \
-    --prefix=/usr --libdir=/usr/lib32
+    --prefix=/usr --libdir=/usr/lib32 --includedir=/usr/include32
   meson compile -C build
 }
 
 package() {
   meson install -C build --destdir "$pkgdir"
-  rm -fr "$pkgdir"/usr/include
   install -Dm644 libdecor-$pkgver/LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
