diff --git a/PKGBUILD b/PKGBUILD
index cfbd961..7f2cff8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,12 +17,12 @@ validpgpkeys=('4A193C06D35E7C670FA4EF0BA2FB9E081F2D130E') #Alan Coopersmith <ala
 validpgpkeys+=('C41C985FDCF1E5364576638B687393EE37D128F8') # Matthieu Herrb <matthieu.herrb@laas.fr>
 
 build() {
-  export CC="gcc -m32"
-  export CXX="g++ -m32"
-  export PKG_CONFIG_PATH="/usr/lib32/pkgconfig"
+  export CC="armv7h-linux-gnueabihf-gcc"
+  export CXX="armv7h-linux-gnueabihf-g++"
+  export PKG_CONFIG="armv7h-linux-gnueabihf-pkg-config"
 
   cd "${srcdir}/libXrender-${pkgver}"
-  ./configure --prefix=/usr --disable-static --libdir=/usr/lib32
+  ./configure --prefix=/usr --disable-static --libdir=/usr/lib32 --includedir=/usr/include32
   make
 }
 
@@ -30,7 +30,7 @@ package() {
   cd "${srcdir}/libXrender-${pkgver}"
 
   make DESTDIR="${pkgdir}" install
-  rm -rf "${pkgdir}"/usr/{include,share,bin}
+  rm -rf "${pkgdir}"/usr/{share,bin}
   mkdir -p "$pkgdir/usr/share/licenses"
   ln -s $_pkgbasename "$pkgdir/usr/share/licenses/$pkgname"
 }
